worlds/jakanddaxter/Rules.py

```diff
@@ -1,102 +1,116 @@
+                          world: JakAndDaxterWorld,
+                          bundle: int) -> bool:
 
-    if options.enable_orbsanity == EnableOrbsanity.option_per_level:
-        bundle_size = options.level_orbsanity_bundle_size.value
-        return can_trade_orbsanity(state, player, bundle_size, required_orbs, required_previous_trade)
-    elif options.enable_orbsanity == EnableOrbsanity.option_global:
-        bundle_size = options.global_orbsanity_bundle_size.value
-        return can_trade_orbsanity(state, player, bundle_size, required_orbs, required_previous_trade)
-    else:
-        return can_trade_regular(state, player, multiworld, required_orbs, required_previous_trade)
+    recalculate_reachable_orbs(state, player)
+    return state.has("Reachable Orbs", player, world.orb_bundle_size * (bundle + 1))
+
+
+def can_reach_orbs_level(state: CollectionState,
+                         player: int,
+                         world: JakAndDaxterWorld,
+                         level_name: str,
+                         bundle: int) -> bool:
 
+    recalculate_reachable_orbs(state, player)
+    return state.has(f"{level_name} Reachable Orbs", player, world.orb_bundle_size * (bundle + 1))
 
-def can_trade_regular(state: CollectionState,
+
+def can_trade_vanilla(state: CollectionState,
                       player: int,
-                      multiworld: MultiWorld,
                       required_orbs: int,
-                      required_previous_trade: int = None) -> bool:
+                      required_previous_trade: typing.Optional[int] = None) -> bool:
 
-    # We know that Orbsanity is off, so count orbs globally.
-    accessible_orbs = can_reach_orbs_global(state, player, multiworld)
+    recalculate_reachable_orbs(state, player)  # With Orbsanity Off, Reachable Orbs are in fact Tradeable Orbs.
     if required_previous_trade:
         name_of_previous_trade = location_table[Cells.to_ap_id(required_previous_trade)]
-        return (accessible_orbs >= required_orbs
-                and state.can_reach(name_of_previous_trade, "Location", player=player))
+        return (state.has("Reachable Orbs", player, required_orbs)
+                and state.can_reach_location(name_of_previous_trade, player=player))
     else:
-        return accessible_orbs >= required_orbs
+        return state.has("Reachable Orbs", player, required_orbs)
 
 
 def can_trade_orbsanity(state: CollectionState,
                         player: int,
-                        orb_bundle_size: int,
                         required_orbs: int,
-                        required_previous_trade: int = None) -> bool:
+                        required_previous_trade: typing.Optional[int] = None) -> bool:
 
-    required_count = math.ceil(required_orbs / orb_bundle_size)
-    orb_bundle_name = orb_item_table[orb_bundle_size]
+    recalculate_reachable_orbs(state, player)  # Yes, even Orbsanity trades may unlock access to new Reachable Orbs.
```

Comment:
What's the case where this is actually possible?

Reply:
You trade 90 Tradeable Orbs to your Uncle, and he gives you the Fisherman's Boat, unlocking Misty Island and it's 150 Reachable Orbs. If seed generation reshuffles the 90th orb to pay your Uncle, then that means Misty Island may not be reachable anymore, and we would recalculate how many orbs are reachable.

Reply:
But seed generation isn't going to make anything unreachable? And even in this scenario, the _Boat_ is what's getting you there in the end, so the Orb shouldn't matter?

Reply:
But if you don't have the 90th orb, you can't pay your uncle for the boat. We can apply the transitive property:

* If I have the 90th orb, I can buy the Boat.
* If I have the Boat, I can reach Misty Island.
* Therefore, if I have the 90th orb, I can reach Misty Island.

Reply:
How are you ever losing the 90th Orb though?